src/EventStore.Core/Services/Transport/Http/Controllers/PersistentSubscriptionController.cs

```diff
@@ -63,6 +65,116 @@ public class PersistentSubscriptionController : CommunicationController {
 				WithParameters(Operations.Subscriptions.ProcessMessages), AckMessages);
 			RegisterUrlBased(service, "/subscriptions/{stream}/{subscription}/nack?ids={messageids}&action={action}",
 				HttpMethod.Post, WithParameters(Operations.Subscriptions.ProcessMessages), NackMessages);
+			//map view parked messages
+			Register(service, "/subscriptions/viewparkedmessages/{stream}/{group}/{event}/{count}?embed={embed}", HttpMethod.Get,
+				ViewParkedMessages, Codec.NoCodecs, AtomCodecs, WithParameters(Operations.Subscriptions.Statistics));
+			Register(service, "/subscriptions/viewparkedmessages/{stream}/{group}?embed={embed}", HttpMethod.Get,
+				ViewParkedMessages, Codec.NoCodecs, AtomCodecs, WithParameters(Operations.Subscriptions.Statistics));
+		}
+		private void ViewParkedMessages(HttpEntityManager http, UriTemplateMatch match) {
+			var stream = match.BoundVariables["stream"];
+			var groupname = match.BoundVariables["group"];
+			var evNum = match.BoundVariables["event"];
+			var cnt = match.BoundVariables["count"];
+
+			stream = "$persistentsubscription-" + stream + "::" + groupname+"-parked";
```

Comment:
We should have other places that build up this key and we should be able to centralize this logic. Just have a look if it's feasible to do this cleanup in this PR.

Reply:
Thanks, added a BuildSubscriptionParkedStreamName which uses previous build function BuildSubscriptionGroupKey